data-prepper-core/src/main/java/com/amazon/dataprepper/plugin/PluginConfigurationConverter.java

```diff
@@ -72,4 +74,11 @@ public Object convert(final Class<?> pluginConfigurationType, final PluginSettin
 
         return configuration;
     }
+
+    private Object convertSettings(final Class<?> pluginConfigurationType, final PluginSetting pluginSetting) {
+        Map<String, Object> settingsMap = pluginSetting.getSettings();
+        if (settingsMap == null)
+            settingsMap = Collections.emptyMap();
```

Comment:
The `PluginSettings` class define the default value for an empty plugin settings over the `PluginConfigurationConverter`

Reply:
Yea, I thought about that as well. But, I didn't want to interrupt any use of `getSettings` that was already using `null`.

From what I can tell, plugins that use the new `pluginConfigurationType` are all assuming non-null configurations. So I wanted to at least cover that.

Reply:
Also, I'd like to fully remove `PluginSettings` someday. :)

Reply:
Thanks for clarifying!